File: carddemo/app/schemas/account.py
"""Account schemas — replaces COACTVW/COACTUP BMS map fields."""

from decimal import Decimal

from pydantic import BaseModel


class AccountBase(BaseModel):
    acct_id: int
    active_status: str = "Y"
    curr_bal: Decimal = Decimal("0.00")
    credit_limit: Decimal = Decimal("0.00")
    cash_credit_limit: Decimal = Decimal("0.00")
    open_date: str = ""
    expiration_date: str = ""
    reissue_date: str = ""
    curr_cyc_credit: Decimal = Decimal("0.00")
    curr_cyc_debit: Decimal = Decimal("0.00")
    addr_zip: str = ""
    group_id: str = ""


class AccountResponse(AccountBase):
    model_config = {"from_attributes": True}


class AccountUpdate(BaseModel):
    active_status: str | None = None
    credit_limit: Decimal | None = None
    cash_credit_limit: Decimal | None = None
    expiration_date: str | None = None
    reissue_date: str | None = None
    addr_zip: str | None = None
    group_id: str | None = None


class AccountListResponse(BaseModel):
    accounts: list[AccountResponse]
    total: int
    page: int
    page_size: int

File: java-migration/domain/src/main/java/com/aws/carddemo/domain/Account.java
package com.aws.carddemo.domain;
import jakarta.persistence.*; import java.math.BigDecimal;
@Entity @Table(name="account")
public class Account {
 @Id @Column(name="acct_id") Long acctId; @Column(name="active_status",length=1) String activeStatus;
 @Column(name="curr_bal",precision=12,scale=2) BigDecimal currBal; @Column(name="credit_limit",precision=12,scale=2) BigDecimal creditLimit; @Column(name="cash_credit_limit",precision=12,scale=2) BigDecimal cashCreditLimit;
 @Column(name="open_date",length=10) String openDate; @Column(name="expiraion_date",length=10) String expiraionDate; @Column(name="reissue_date",length=10) String reissueDate;
 @Column(name="curr_cyc_credit",precision=12,scale=2) BigDecimal currCycCredit; @Column(name="curr_cyc_debit",precision=12,scale=2) BigDecimal currCycDebit; @Column(name="addr_zip",length=10) String addrZip; @Column(name="group_id",length=10) String groupId;
}
